doc_src/terminal-compatibility.rst

```diff
@@ -0,0 +1,179 @@
+Terminal Compatibility
+======================
+
+To make full use of your terminal, fish emits a number of control sequences and control characters.
+
+The terminal must be able to parse Control Sequence Introducer (CSI) commands, Operating System Commands (OSC) and optionally Device Control Sequences (DCS).
+These are defined by ECMA-48.
+If a valid CSI, OSC or DCS sequence does not represent a command implemented by the terminal, the terminal will ignore it.
+
+Here are the commands used by fish.
+
+Control sequences are denoted in a fish-like syntax:
+special characters like ``[`` and ``;`` are not escaped (except for backslash); spaces are only added for readability.
+Placeholders are written in angle brackets: ``<Ps>`` for a number and ``<Pt>`` for arbitrary printable strings.
+
+Required Commands
+-----------------
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Sequence
+     - Description
+     - Origin
+   * - ``\n``
+     - move cursor down one line (Terminfo: ``cursor_down`` / ``cud1``)
+     -
+   * - ``\e[ <Ps> A``
+     - move cursor up <Ps> times (Terminfo: ``parm_up_cursor`` / ``cuu``)
+     - VT100
+   * - ``\e[ <Ps> C``
+     - move cursor to the right <Ps> times (Terminfo: ``parm_right_cursor`` / ``cuf``)
+     - VT100
+   * - ``\e[ <Ps> D``
+     - move cursor to the left <Ps> times (Terminfo: ``parm_left_cursor`` / ``cub``)
+     - VT100
+   * - ``\e[H``
+     - set cursor position (move to row 1, column 1) (Terminfo: ``cursor_address`` / ``cup``)
+     - VT100
+   * - ``\e[K``
+     - clear to end of line (Terminfo: ``clr_eol`` / ``el``)
+     - VT100
+   * - ``\e[J``
+     - clear to the end of screen (Terminfo: ``clr_eos`` / ``ed``)
+     - VT100
+   * - ``\e[2J``
+     - clear the entire screen (Terminfo: ``clear_screen`` / ``clear``)
+     - VT100
+   * - ``\e[0c``
+     - request primary device attribute. The terminal must respond. An example response is ``\e[?62;4;22;28c``.
```

Comment:
Do we care about the contents of the response?

Reply:
no, I changed it to state only what we require (`CSI ?` prefix and `c` suffix).
I originally gave a real-world example because the parameters indicate capabilities
so just `CSI ? 0 c` might be wrong. But I guess that's not our problem,
people can see refer to the XTerm ctlseqs.html